tlog-cosignature.md

```diff
@@ -71,34 +80,57 @@ over the message defined in the next section.
+The signed message MUST be three newline (U+000A) terminated lines (one header
+line, one name line, and one timestamp line) followed by the whole note body of
+the cosigned checkpoint (including the final newline, but not including any
+signature lines).
 
-The header line MUST be the fixed string `cosignature/v1`, and provides domain
+The header line MUST be the fixed string `cosignature/v2`, and provides domain
 separation.
 
+The name line MUST be the same as in the signed-note signature line, and ensures
+the witness name is cryptographically tied to the resulting cosignature.
+
 The timestamp line MUST consist of the string `time`, a single space (0x20), and
 the number of seconds since the UNIX epoch encoded as an ASCII decimal with no
 leading zeroes. This value MUST match the `timestamped_signature.timestamp`.
 
     cosignature/v1
+    witness.example.com/w1
     time 1679315147
     example.com/behind-the-sofa
     20852163
     CsUYapGGPo4dkMgIAUqom/Xajj7h2fB2MPA3j2jxq2I=
 
-Semantically, a v1 cosignature is a statement that, as of the specified time,
+Semantically, a v2 cosignature is a statement that, as of the specified time,
 the consistent tree head with the largest size the witness has observed for the
 log identified by the origin line has the specified root hash.
 
 Extension lines MAY be included in the checkpoint by the log, and if present
 MUST be included in the cosigned message. However, it's important to understand
 that the witness is asserting observation of correct append-only operation of
-the log based on the first three lines of the checkpoint; no semantic statement
-is made about any extension lines, and consensus between witnesses on the
-extension lines SHALL NOT be assumed.
+the log based on the first three lines of the checkpoint; consensus between
+witnesses on the extension lines SHALL NOT be assumed, and no semantic statement
+is made about any extension lines unless the witness operator says otherwise.
+
+A witness operator MAY make additional statements about a checkpoint.  These
+additional statements need to be communicated out of band to those defining
+trust policies based on tuples of (witness name, public key, cosignature/v2).
+The same public key can be used with different witness names for disambiguation.
+
+Examples of additional statements include: "I also mirrored the log up until the
+checkpoint tree size" and "I also reproduced the sparse Merkle tree hash in the
+extension lines for this checkpoint size".  See [tlog-mirror][] for an example.
+
+## Changes compared to cosignature/v1
+
+* The witness name is cryptographically tied to the resulting cosignature.
+* It is explicitly allowed to make additional statements about a checkpoint.
+  These additional statements need to be communicated out of band, identified by
+  the tuple (witness name, public key, cosignature/v2) in trust policies.  In
+  cosignature/v1, the public key cannot be reused with different witness names.
```

Comment:
Can you add some justification for _why_ these changes are useful. The main purpose behind this seems to be to allow the same key to be used in different contexts (by setting different witness names). Why is embedding the name better than using different keys for different types of claim?